Sources/firebase/include/FirebaseFirestore.hh

```diff
@@ -64,6 +64,11 @@ snapshot_reference(const ::firebase::firestore::DocumentSnapshot snapshot) {
   return snapshot.reference();
 }
 
+inline bool
+snapshot_exists(const ::firebase::firestore::DocumentSnapshot snapshot) {
```

Comment:
Please take this by const reference to avoid a copy.

```suggestion
snapshot_exists(const ::firebase::firestore::DocumentSnapshot &snapshot) {
```